crates/circuit/src/operations.rs

```diff
@@ -286,6 +288,27 @@ impl Operation for OperationRef<'_> {
             Self::Unitary(unitary) => unitary.directive(),
         }
     }
+    #[inline]
+    fn get_mat_static_1q(&self, params: &[Param]) -> Option<[[Complex64; 2]; 2]> {
+        match self {
+            Self::StandardGate(standard) => standard.get_mat_static_1q(params),
+            Self::StandardInstruction(instruction) => instruction.get_mat_static_1q(params),
+            Self::Gate(gate) => gate.get_mat_static_1q(params),
+            Self::Instruction(instruction) => instruction.get_mat_static_1q(params),
+            Self::Operation(operation) => operation.get_mat_static_1q(params),
+            Self::Unitary(unitary) => unitary.get_mat_static_1q(params),
+        }
+    }
+    fn get_mat_nalgebra_1q(&self, params: &[Param]) -> Option<Matrix2<Complex64>> {
```

Comment:
I don't see the `nalgebra` function being used anywhere -- is there a plan for using this somewhere? Otherwise I'm usually in favor of not adding this, as it's also a path that is not tested 🤔

Reply:
I didn't put in initially, but Matthew has asked for it :). Anyhow, I have locally double-checked that the function works (modifying the ``euler_one_qubit_decomposer`` to work with nalgebra matrices instead). In general, we probably need to start adding more pure rust tests.